controllers/awschaos/detachvolume/types.go

```diff
@@ -0,0 +1,136 @@
+// Copyright 2020 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package detachvolume
+
+import (
+	"context"
+	"errors"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+
+	"github.com/chaos-mesh/chaos-mesh/api/v1alpha1"
+	"github.com/chaos-mesh/chaos-mesh/pkg/router"
+	ctx "github.com/chaos-mesh/chaos-mesh/pkg/router/context"
+	end "github.com/chaos-mesh/chaos-mesh/pkg/router/endpoint"
+)
+
+const (
+	AwsFinalizer = "aws-Finalizer"
+)
+
+type endpoint struct {
+	ctx.Context
+}
+
+func (e *endpoint) Apply(ctx context.Context, req ctrl.Request, chaos v1alpha1.InnerObject) error {
+	awschaos, ok := chaos.(*v1alpha1.AwsChaos)
+	if !ok {
+		err := errors.New("chaos is not awschaos")
+		e.Log.Error(err, "chaos is not AwsChaos", "chaos", chaos)
+		return err
+	}
+
+	secret := &v1.Secret{}
+	err := e.Client.Get(ctx, types.NamespacedName{
+		Name:      awschaos.Spec.SecretName,
+		Namespace: awschaos.Namespace,
+	}, secret)
+	if err != nil {
+		e.Log.Error(err, "fail to get cloud secret")
+		return err
+	}
+
+	ec2client := ec2.New(ec2.Options{
+		Region:      awschaos.Spec.AwsRegion,
+		Credentials: credentials.NewStaticCredentialsProvider(string(secret.Data["aws_access_key_id"]), string(secret.Data["aws_secret_access_key"]), ""),
```

Comment:
aws_access_key_id and aws_secret_access_key should be constants.

this required static credentials, it's not safe. it would be better to use static credentials when secret not nils. otherwise use default AWS credentials chains.

Reply:
If it is not obtained through the secret file, it may only be possible to obtain the access_key through the local environment variable?

Reply:
for example, the ec2 instance supports ec2 meta. and EKS support web identity provider. they do not need to use static credentials.

Reply:
oh, let me try it.